add_source_file

```diff
@@ -151,26 +156,31 @@ def add_to_source_list(filename, source_list):
     if filename.suffix != ".cpp":
         return
 
-    # if the target already has an entry with the same filename, loudly skip
-    if file_line in sl_lines:
-        print(f"  '{filename}' already found in '{source_list}', skipping")
-        return
+    if remove:
+        if file_line in sl_lines: sl_lines.remove(file_line)
+    else:
+        # if the target already has an entry with the same filename, loudly skip
+        if file_line in sl_lines:
+            print(f"  '{filename}' already found in '{source_list}', skipping")
+            return
+
+        sl_lines.append(file_line)
 
-    sl_lines.append(file_line)
     sl_lines.sort()
     open(source_list_file, 'w').writelines(sl_lines)
 
-def add_to_source_lists(filename, targets):
-    translated_targets = [source_list_target_translations[t] for t in targets]
+def add_to_source_lists(filename, targets, remove):
+    # remove from all tagerts if -r was specified.
+    translated_targets = source_list_target_translations.values() if remove else [source_list_target_translations[t] for t in targets]
```

Comment:
Having a separate `def remove_from_source_lists(filename, targets)` would be easier to read than this line, even if it ended up calling `add_to_source_list(filename, t, True)`.

Reply:
then the code im main would havea less clean structure tho :s

Reply:
The code in main currently prints "adding filename" even when removing, and sanity checks that the files exist even when removing. I'd describe it as buggy rather than clean.

Reply:
that's unrelated to this suggested change, but something that should probably be fixed

Reply:
Fixing it also alters the reasoning about whether the suggested change should be done.